Added video

lucascumsille opened the pull request:
### Added video "What we can and  can't fix" to homepage.
This part shouldn't need any further work.
<img width="1428" alt="Screenshot 2022-02-22 at 08 27 02" src="https://user-images.githubusercontent.com/13790153/155333296-f121c5e4-051b-4471-9a1b-cce9499e49af.png">


### Added video when reporting and selecting category
Pending for developer: Needs a conditional in one of the templates where the video is only displayed for Oxfordshire. 
<img width="1428" alt="Screenshot 2022-02-22 at 08 26 50" src="https://user-images.githubusercontent.com/13790153/155333665-15c13cee-62d6-4c9b-bd6a-569848c8f096.png">

### Added video "Why we don't fix things" to Oxfordshire email template. 
@dracos I didn't place the video under the report details, It wouldn't be very visible, especially on mobile, where is going to sit right at the bottom of the email. I'm guessing the council really want the user to watch the video, so they don't call complaining straight away. So I placed the video right under the "Closed" status, so the user will see that the issue wasn't fixed, but they will have something explaining why not straight away.
Pending for developer: I added the conditional, but please test it first.
<img width="624" alt="Screenshot 2022-02-23 at 13 41 13" src="https://user-images.githubusercontent.com/13790153/155333979-d164a237-0c71-44fb-b4cc-3abab62681d2.png">

Fixes part of: [MySociety/societyworks/2831](https://github.com/mysociety/societyworks/issues/2831)

Please let me know if you have any feedback. [skip changelog]

chrismytton commented on templates/web/oxfordshire/report-video.html:
```diff
@@ -0,0 +1,9 @@
+<div class="video-thumbnail-section">
+    <a class="iframe fancybox" rel="fancy" href="https://www.youtube.com/embed/2mxbgJXGHTQ">
+        <img class="video-thumbnail" alt="What we can and can’t fix video" src="/cobrands/oxfordshire/images/oxfordshire-can-cants-video.png">
+        <div class="video-cap-wrapper">
+            <p style="color:#222"><strong>Watch</strong></p>
+            <p>Cllr Tim Bearder explains what we can and can’t fix</p>
+        </div>
+    </a>
+</div>
```
This template doesn't seem to be included anywhere?

lucascumsille replied:
Thanks for the feedback @chrismytton. About this I'm not sure where to include the conditional that test IF Oxfordshire then include the video. 
At the beginning I was going to include it with Jquery, but with Moray we thought would be better to include in a template and then add a conditional to test if the council is Oxfordshire.

dracos replied:
I think if you move report-video.html to be report/new/top_message.html does that then appear where you would like? That is a cobrand-specific template for a top message.

lucascumsille replied:
@dracos yes that's perfect, Matthew =)

chrismytton commented on web/cobrands/oxfordshire/js.js:
```diff
@@ -75,4 +75,3 @@
         }
     }
 })();
-//*/
```
I think this was put here on purpose, there's a similar one at the start of the file. I think they should either both be removed or both left in.

chrismytton commented on templates/web/oxfordshire/index-steps.html:
```diff
@@ -1,3 +1,9 @@
+<h2>[% loc('What we can and can’t fix') %]</h2>
+
+<div class="embed-responsive embed-responsive-16by9">
+    <iframe class="embed-responsive-item" src="https://www.youtube.com/embed/2mxbgJXGHTQ" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
On the [Freshdesk ticket](https://mysocietysupport.freshdesk.com/a/tickets/1648) I asked Oxfordshire if they were going to put the video on an official account, as this video is currently on Tom's personal account. Their response was "I'm happy to use whatever works best for you guys. YouTube, Vimeo? or just hosted on a server somewhere. Happy for your technical needs to dictate where it's held.".

So I think we need to send Oxfordshire a message via the Freshdesk thread asking them to uploads the videos to an official Oxfordshire County Council YouTube channel, then once that's done we can update these embeds to point at the new videos.

lucascumsille replied:
Cool, I'll ask @lizettal for an account, because I don't have one on fresh desk.

lucascumsille replied:
Great, Lizetta will send a message to the council.

lizettal replied:
@jacquelinebylau can you please follow up with the council on this as just mentioned

jacquelinebylau replied:
@lucascumsille 
Here are the links:
Intro: https://youtu.be/Uf0b_JAzpcY
why we haven't fixed: https://youtu.be/hae9g170sA4

lucascumsille replied:
Thanks @jacquelinebylau. Appreciate it!

chrismytton reviewed (changes requested):
Looking good! Just a few tweaks needed I think.

dracos commented on templates/web/oxfordshire/index-steps.html:
```diff
@@ -1,3 +1,9 @@
+<h2>[% loc('What we can and can’t fix') %]</h2>
+
+<div class="embed-responsive embed-responsive-16by9">
+    <iframe class="embed-responsive-item" src="https://www.youtube.com/embed/Uf0b_JAzpcY" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Shuold we use youtube-nocookie.com (or whatever it is) to avoid a cookie?

lucascumsille replied:
Thanks for that, I didn't know it. I changed the embed so they use that one instead.

dracos commented on templates/email/oxfordshire/alert-update.html:
```diff
@@ -0,0 +1,34 @@
+[% WRAPPER '_email_sidebar.html' object = report %]
+    <h2 style="[% h2_style %]">[% title %]</h2>
+    <p style="[% secondary_p_style %]">[% report.detail | html %]</p>
+    [% TRY %][% INCLUDE '_council_reference_alert_update.html' problem=report p_style=secondary_p_style %][% CATCH file %][% END %]
```
Do we want the video at the bottom of the left column, or could it go the bottom of the right column? If it can go bottom of the right, you don't need to override alert-update.html/txt, you can put the video in this template and then it will be included there.

lucascumsille replied:
@dracos This was my reasoning to place the video at the bottom of the left column:

I didn't place the video under the report details, It wouldn't be very visible, especially on mobile, where is going to sit right at the bottom of the email. I'm guessing the council really want the user to watch the video, so they don't call complaining straight away. So I placed the video right under the "Closed" status, so the user will see that the issue wasn't fixed, but they will have something explaining why not straight away.

dracos replied:
Okay, though there might be more than one update conceivably :) (but probably not often.) That's fine, but what we should do is add a new INCLUDE template at the place you want to put the video so that you don't need to override the whole alert-update template, but instead can add an oxfordshire specific template of just the needed bit, similar to how the include is done here.

`_alert_update_after_updates.html` would be the first name I think of.

lucascumsille replied:
Awesome I just push a commit for with this change.

dracos commented on templates/email/oxfordshire/alert-update.txt:
```diff
@@ -0,0 +1,32 @@
+Subject: New [% site_name %] updates on report: '[% report.title %]'
```
This txt template does not include a link to the video anywhere. But see my other comment about which template to use - if that comment is approved, you could put the link in the equivalent _council_reference_alert_update.txt

lucascumsille replied:
@dracos see comment above. But I wasn't sure what to do with this file, I noticed when I didn't include it I was getting an error when loading the email on my browser.

dracos replied:
It's for the text part of the email - for clients that don't handle HTML email, you need to include all the contents from the HTML part here also. So just a link to the YouTube video somewhere would presumably be enough.

lucascumsille replied:
Thanks for clarifying, Matthew.  I just included some info + the link.

dracos commented on templates/web/oxfordshire/report-video.html:
```diff
@@ -0,0 +1,9 @@
+<div class="video-thumbnail-section">
+    <a class="iframe fancybox" rel="fancy" href="https://www.youtube.com/embed/Uf0b_JAzpcY">
```
I guess, though I'd rarely suggest this, opening in a new tab might be wanted/expected here, given you are mid reporting flow.

lucascumsille replied:
@dracos this is one is using a Lightbox(fancybox) to watch the video so we are not leaving the page.

dracos reviewed (changes requested):
One query about video location in email, one thought about the domain for the embedded one, and one about opening in a new tab :)

dracos commented on templates/email/default/alert-update.html:
```diff
@@ -14,6 +14,7 @@
   [% start_padded_box | safe %]
   <h1 style="[% h1_style %]">New updates on <a href="[% problem_url %]">[% title %]</a></h1>
   [%~ INCLUDE '_email_comment_list.html' %]
+  [%~ INCLUDE '_alert_update_after_updates.html' %]
```
This needs a TRY/CATCH/END wrapper around it like the other INCLUDE below in the same file, otherwise it will error in other cobrands where the template does not exist.

dracos commented on templates/email/oxfordshire/_alert_update_after_updates.txt:
```diff
@@ -0,0 +1,2 @@
+Watch video Cllr Tim Bearder explains when we don’t fix things
```
This is good, although in the text part single newlines will be smushed together in a text email into one paragraph, so you might want to have "Watch video Cllr Tim Bearder explains when we don’t fix things: https://youtu.be/hae9g170sA4"

Also, you need to include this template in the default/alert-update.txt file :) (also with the TRY/CATCH/END).

dracos reviewed (changes requested):
Almost, just needs a couple of changes for the new includes :)

lucascumsille commented:
@dracos Thanks for all the feedback, Matthew. Much appreciated 
I just added the latest comments you included. =)

lucascumsille commented:
Hi @dracos do you think this PR ready to be merged to master?

dracos commented:
> Hi @dracos do you think this PR ready to be merged to master?

For things like this, we put it on staging first for them to approve before merging in case they want any changes. I'll check your latest changes this morning, I'm sure they'll be fine, I'll put it on staging and let them know.

lucascumsille commented:
> > Hi @dracos do you think this PR ready to be merged to master?
> 
> For things like this, we put it on staging first for them to approve before merging in case they want any changes. I'll check your latest changes this morning, I'm sure they'll be fine, I'll put it on staging and let them know.

Makes sense, thanks for that Matthew. Let me know if you need any further changes.

dracos reviewed (commented):
The front page and email work nicely - though do we want a bit of padding under the video on the front page? The new report video though doesn't open in a fancybox, it opens in a new tab full screen, I'm just looking into why.

lucascumsille commented:
> 



> The front page and email work nicely - though do we want a bit of padding under the video on the front page? The new report video though doesn't open in a fancybox, it opens in a new tab full screen, I'm just looking into why.

That's weird fancy box was working when I had added the whole snippet using Jquery. 
Btw I just added extra space to the video in the homepage

dracos commented:
It's because the left column gets pulled in via Ajax upon map click/page load, and the fancybox handler was only being set up on page load, not after. I've added a `fixmystreet.set_up.fancybox_images()` call to the function that deals with when data is returned for the left column.

lucascumsille commented:
> It's because the left column gets pulled in via Ajax upon map click/page load, and the fancybox handler was only being set up on page load, not after. I've added a `fixmystreet.set_up.fancybox_images()` call to the function that deals with when data is returned for the left column.

Amazing, thank you. It's working well for me. I just added commit that fixes a size style issue on the reporting a new problem page for IE11.
